plugins/jobs/web_client/views/CheckBoxMenu.js

```diff
@@ -1,27 +1,63 @@
+import _ from 'underscore';
 import View from 'girder/views/View';
 import JobCheckBoxMenuTemplate from '../templates/jobCheckBoxMenu.pug';
+import jobCheckBoxContentTemplate from '../templates/jobCheckBoxContent.pug';
 
 var CheckBoxMenu = View.extend({
     events: {
-        'click input': function (e) {
-            var checkBoxStates = {};
-            $.each(this.$('input'), function (i, input) {
-                checkBoxStates[input.id] = input.checked;
+        'click input.g-job-filter-checkbox': function (e) {
+            e.stopPropagation();
+            _.keys(this.values).forEach(key => {
+                if (e.target.id === key) {
+                    this.values[key] = e.target.checked;
+                }
             });
-            this.trigger('g:triggerCheckBoxMenuChanged', checkBoxStates);
+            this.checkAllChecked = this.allItemChecked();
+            this._renderContent();
+            this.trigger('g:triggerCheckBoxMenuChanged', this.values);
+        },
+        // When a label wrap a input and the label is clicked, there will be two event being triggered, stop the first one from bubbling up to prevent unexpected behavior
+        'click label': function (e) {
+            e.stopPropagation();
+        },
+        'click .dropdown-menu': e => {
+            e.stopPropagation();
+        },
+        'click .g-job-checkall input': function (e) {
+            e.stopPropagation();
+            var checked = !this.allItemChecked();
+            this.checkAllChecked = checked;
+            _.keys(this.values).forEach(key => {
+                this.values[key] = checked;
+            });
+            this._renderContent();
+            this.trigger('g:triggerCheckBoxMenuChanged', this.values);
```

Comment:
I just kept the logic of 'g:triggerCheckBoxMenuChanged'. Is this event not scoped to only its subscriber? Extend values sounds like a good idea, will the event be emitted automatically when values change?